src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -516,19 +425,175 @@ public void TypeDeclaration_IsStruct()
             var src = @"
 record struct Point(int x, int y);
 ";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+
+            CompileAndVerify(comp, symbolValidator: validateModule, sourceSymbolValidator: validateModule);
+
+            static void validateModule(ModuleSymbol module)
+            {
+                var isSourceSymbol = module is SourceModuleSymbol;
+
+                var point = module.GlobalNamespace.GetTypeMember("Point");
+                Assert.True(point.IsValueType);
+                Assert.False(point.IsReferenceType);
+                Assert.False(point.IsRecord);
+                Assert.Equal(TypeKind.Struct, point.TypeKind);
+                Assert.Equal(SpecialType.System_ValueType, point.BaseTypeNoUseSiteDiagnostics.SpecialType);
+                Assert.True(SyntaxFacts.IsTypeDeclaration(SyntaxKind.RecordStructDeclaration));
```

Comment:
I'm confused by this assert. It seems independent of any particular compilation, but is rather just testing the SyntaxKind constant. Was that intentional?

Reply:
Yes, it's just testing that SyntaxFacts API :-) 
Note that although it appears as a change in the diff, this is an existing check.

---
In reply to: [606013588](https://github.com/dotnet/roslyn/pull/52256#discussion_r606013588) [](ancestors = 606013588)

Reply:
I'll move it out of the `validateModule` local function

---
In reply to: [606392703](https://github.com/dotnet/roslyn/pull/52256#discussion_r606392703) [](ancestors = 606392703,606013588)